toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,236 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
-import java.util.Properties;
+import java.net.UnknownHostException;
+import java.util.Map;
 
 import javax.imageio.ImageIO;
 
-import org.apache.commons.io.FileUtils;
+import io.toast.tk.agent.ui.i18n.CommonMessages;
+import io.toast.tk.agent.ui.i18n.MainAppMessages;
+import io.toast.tk.agent.ui.provider.ConfigPanelProvider;
+import io.toast.tk.agent.ui.provider.PropertiesProvider;
+import io.toast.tk.agent.ui.verify.IPropertyVerifier;
+import io.toast.tk.core.annotation.ActionAdapter;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 import com.google.inject.Inject;
 
-import io.toast.tk.agent.config.WebConfig;
-import io.toast.tk.agent.config.WebConfigProvider;
+import io.toast.tk.agent.config.AgentConfig;
+import io.toast.tk.agent.config.AgentConfigProvider;
 import io.toast.tk.agent.web.BrowserManager;
 import io.toast.tk.agent.web.IAgentServer;
 import io.toast.tk.agent.web.RestRecorderService;
 
 public class MainApp implements IAgentApp {
 
 	private static final Logger LOG = LogManager.getLogger(MainApp.class);
-	
-	final static String PROPERTY_FILE = "/toast.web.properties";
+
+	private final ConfigPanelProvider configPanelProvider;
 	private BrowserManager browserManager;
-	private File toastWebPropertiesFile;
-	private WebConfigProvider webConfigProvider;
-	private final Properties webProperties;
+	private AgentConfigProvider webConfigProvider;
+	private final PropertiesProvider propertiesProvider;
 	private TrayIcon trayIcon;
-	private Image online_image;
-	private Image offline_image;
+	private Image onlineImage;
+	private Image offlineImage;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the "offlineImage" field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1450)